🌟 **Exercise 4** : Forward Propagation Calculation

In this exercise, you will manually compute the forward propagation of a simple neural network that predicts house prices based on:

Square Footage (x₁)

Number of Bedrooms (x₂)

In [1]:
# We will calculate the output using the following given values:

# Input Values:
#     x₁ = 2000 (Square Footage)
#     x₂ = 3 (Number of Bedrooms)
# Initial Weights:
#     w₁ = 0.5 (Weight for Square Footage)
#     w₂ = 0.7 (Weight for Bedrooms)
# Bias: b = 50,000
# Activation Function: ReLU (Rectified Linear Unit)

1. **Calculate the output value “z” before activation.**

In [2]:
x1 = 2000   # square footage
x2 = 3      # nb of bedrooms

# Initial weight
w1 = 0.5
w2 = 0.7

# Bias
b = 50000

# Calculate the output value 'z' before activation
z = x1 * w1 + x2 * w2 + b

2. **Apply the ReLU function to compute the final prediction.**

In [3]:
reLU = max(0, z)
print(f"The final prediction is: {reLU} $")

The final prediction is: 51002.1 $


3. **Interpret the result: What is the predicted house price?**


The ReLU activation function returns the value of the weighted sum if it is positive, otherwise it returns 0.

In this case, the result is 51,002.10, which means the predicted house price is $51,002.10.